### ISSUE TITLE
interactive switch -i has no effect if script is terminated by exception

### ISSUE THREAD

**Issue by skjaeve** [ipython/ipython#9315]
IPython version 4.1.2, installed with `pip install --user --upgrade ipython`.

Example script:

    raise Exception('bug in ipython')

run it with -i switch:

```
$ ipython -i ipython-test.py 
Python 2.7.10 (default, Sep  8 2015, 17:20:17) 
Type "copyright", "credits" or "license" for more information.

IPython 4.1.2 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
/src/ipython-test.py in <module>()
----> 1 raise Exception('bug in ipython')

Exception: bug in ipython
$ 

```
Expected behavior: Exception, then Ipython prompt
Actual behavior: Exception, then return to Bash prompt

This bug is new in IPython version 4. Behavior is identical in Python2 and Python3 versions of IPython.

**Comment by jkfurtney** [comment]
The behavior is correct for me in IPython 4.0.0 but I get this bug in IPython 4.1.2.

**Comment by takluyver** [comment]
See #9223 and #9368. This was fixed in #9225.

@minrk I reckon we need to cut a release with that fix.

**Comment by minrk** [comment]
Yup, we have lots of releasing to do. Let's try to get this one and a few of the others out this week.
